File: book/src/schema/tuples/basic.go
package main

func main() {}

type Tuple2[T1 any, T2 any] struct {
	F1 T1
	F2 T2
}

/*
type Tuple3[T1 any, T2 any, T3 any] struct {
	F1 T1
	F2 T2
	F2 T3
}

...and so on...
*/

type Sample struct {
	Size Tuple2[uint32, uint32]
}

File: book/src/reference/schema/tuples/basic.kt
fun main() {}

// #region snippet
data class Tuple2<T1, T2>(
    val f1: T1,
    val f2: T2,
)

/*
data class Tuple3<T1, T2, T3>(
    val f1: T1,
    val f2: T2,
    val f3: T3
)

...and so on...
*/

data class Sample(
    val size: Tuple2<UInt, UInt>,
)
// #endregion snippet
